recipes/system-probe.rb

```diff
@@ -0,0 +1,58 @@
+#
+# Cookbook Name:: datadog
+# Recipe:: system-probe
+#
+# Copyright 2011-2019, Datadog
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# Set the correct agent startup action
+sysprobe_agent_start = node['datadog']['system_probe']['enabled'] ? :start : :stop
+
+#
+# Configures system-probe agent
+system_probe_config_file = ::File.join(node['datadog']['agent6_config_dir'], 'system-probe.yaml')
+template system_probe_config_file do
+  source 'system_probe.yaml.erb'
+  variables(
+    enabled: node['datadog']['system_probe']['enabled'],
+    sysprobe_socket: node['datadog']['system_probe']['sysprobe_socket'],
+    debug_port: node['datadog']['system_probe']['debug_port'],
+    bpf_debug: node['datadog']['system_probe']['bpf_debug'],
+    enable_conntrack: node['datadog']['system_probe']['enable_conntrack']
+  )
+  owner 'dd-agent'
+  group 'root'
+  mode '640'
+  notifies :restart, 'service[datadog-agent-sysprobe]', :delayed unless sysprobe_agent_start == false
```

Comment:
The tests on `sysprobe_agent_start` don't look consistent:
 - it is set to `false` in `default/attributes.rb` (boolean)
 - line 21 you set it to either `:start` or `:stop` (Symbol)
 - here, you're comparing it to a boolean

Reply:
I meant to use `node['datadog']['system_probe']['enabled'] ==false`, somehow I put `sysprobe_agent_start` in there......